runtime/lua/vim/lsp/buf.lua

```diff
@@ -192,17 +192,21 @@ end
 ---     - range (table|nil) Range to format.
 ---         Table must contain `start` and `end` keys with {row,col} tuples using
 ---         (1,0) indexing.
+---         Can also be a list of tables that contain `start` and `end` keys as described above,
+---         in which case textDocument/rangesFormatting support is required.
 ---         Defaults to current selection in visual mode
 ---         Defaults to `nil` in other modes, formatting the full buffer
 function M.format(options)
   options = options or {}
   local bufnr = options.bufnr or api.nvim_get_current_buf()
   local mode = api.nvim_get_mode().mode
   local range = options.range
-  if not range and mode == 'v' or mode == 'V' then
+  local ranges = (range and #range ~= 0 and type(range[1]) == 'table') and range
```

Comment:
Maybe it's just me but I find this code that deals with both scenarios of dealing with range vs ranges a bit hard to follow. I would suggest either adding some comments that explain what's going on, or clearly separating both cases with a single if-statement.

Reply:
Good point, having both `range` and `ranges` as a variable is confusing - I added some comments, but also replaced the `ranges` variable with a boolean `passed_multiple_ranges`, I think readability is much higher now